### PR TITLE
Create metric for time taken by validator to perform duties

### LINKED ISSUE #6354: Create metric for time taken by validator to perform duties
Basically, when the validator calls to the BN, capture the time taken for the duty. this would include getting duties and performing duties.

Related classes/interfaces:
- Duty
- AttestationProductionDuty
- SlotBasedScheduledDuties

We could report those metrics similar to the way we do in BlockImportMetrics.

Requirements:
- Should be enabled when the validator metrics flag is enabled
- It should work for Remote or Embedded VC
- Metric should be on the `validator` namespace

### PR DESCRIPTION
**Pull request by courtneyeh**
fixes #6354 

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/duties/ValidatorDutyMetrics.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import org.hyperledger.besu.plugin.services.MetricsSystem;
+import org.hyperledger.besu.plugin.services.metrics.LabelledMetric;
+import org.hyperledger.besu.plugin.services.metrics.OperationTimer;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.metrics.TekuMetricCategory;
+
+public class ValidatorDutyMetrics {
+  private final LabelledMetric<OperationTimer> dutyMetric;
+
+  ValidatorDutyMetrics(final LabelledMetric<OperationTimer> dutyMetric) {
```
Comment (lucassaldanha) [inline validator/client/src/main/java/tech/pegasys/teku/validator/client/duties/ValidatorDutyMetrics.java]: Can we make this private?

File: validator/client/src/test/java/tech/pegasys/teku/validator/client/duties/ValidatorDutyMetricsTest.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    doReturn(timingContext).when(operationTimer).startTimer();
+
+    validatorDutyMetrics = ValidatorDutyMetrics.create(metricsSystem);
+  }
+
+  @Test
+  public void shouldRecordDutyTimeWhenDutySucceeds() throws Exception {
+    when(duty.performDuty()).thenReturn(SafeFuture.completedFuture(DutyResult.NO_OP));
+
+    validatorDutyMetrics.performDutyWithMetrics(duty).get();
```
Comment (lucassaldanha) [inline validator/client/src/test/java/tech/pegasys/teku/validator/client/duties/ValidatorDutyMetricsTest.java]: We can use:
```
assertThatSafeFuture(validatorDutyMetrics.performDutyWithMetrics(duty)).isCompleted();
```

(and also remove the `throws exception` from the method)

**Review by lucassaldanha — APPROVED** [review]

**Review by rolfyone — COMMENTED** [review]
Don't forget a changelog entry :)

**Review by rolfyone — APPROVED** [review]
Looks good!

**Pull request merged**